File: Backend API/CleaningHouse_API/Models/Customers/UserLocation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CleaningHouse_API.Models.Common;

namespace CleaningHouse_API.Models.Customers;

[Table("UserLocations")]
public class UserLocation
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int UserId { get; set; }

    [Required]
    [MaxLength(200)]
    public string LocationName { get; set; } = string.Empty;

    [Required]
    public double Lat { get; set; }

    [Required]
    public double Lng { get; set; }

    public bool IsActive { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey(nameof(UserId))]
    public virtual AppUser? AppUser { get; set; }

    public virtual ICollection<Booking>? Bookings { get; set; }
}

File: App_Clint/lib/features/user_locations/domain/entities/user_location.dart
import 'package:equatable/equatable.dart';

class UserLocation extends Equatable {
  final int id;
  final int userId;
  final String locationName;
  final double lat;
  final double lng;
  final DateTime createdAt;

  const UserLocation({
    required this.id,
    required this.userId,
    required this.locationName,
    required this.lat,
    required this.lng,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [id, userId, locationName, lat, lng, createdAt];
}
